lib/ui/painting/image_descriptor.cc

```diff
@@ -56,6 +73,15 @@ ImageDescriptor::ImageDescriptor(sk_sp<SkData> buffer,
           static_cast<SkCodecImageGenerator*>(
               SkCodecImageGenerator::MakeFromCodec(std::move(codec))
                   .release()))),
+      platform_image_generator_(nullptr),
+      image_info_(CreateImageInfo()),
```

Comment:
The order of field initialization matters here since CreateImageInfo() relies on `generator_` and `platform_image_generator_` being set.  That is fragile and actually not declared in the c++ spec if I remember correctly.  It's better to be safe than sorry and call CreateImageInfo in the constructor.

Reply:
If you change the orders of these, clang will yell at you. I'm doing this to avoid the overhead of initializing the struct twice. The spec says:

```
1) If the constructor is for the most-derived class, virtual bases are initialized in the order in which they appear in depth-first left-to-right traversal of the base class declarations (left-to-right refers to the appearance in base-specifier lists)
2) Then, direct bases are initialized in left-to-right order as they appear in this class's base-specifier list
3) Then, non-static data member are initialized in order of declaration in the class definition.
4) Finally, the body of the constructor is executed
```

So I think this is ok.

I think if anything this is pointing to refactoring this class into two or three separate classes, but that adds different mental overhead to deal with :\

Reply:
`non-static data member are initialized in order of declaration in the class definition.`  The order of the definitions has to match those of the declarations, otherwise it's just a warning.  The order in the declarations know nothing about the order that the definitions should be.

I think it's safer to just enforce the order in the constructor instead of relying on the nooks and crannies of the c++ spec and relying that a warning is enough to stop breakages.

I think you can avoid the double initialization by not initializing the field then setting it.  All things considered I'll leave it to you to decide what you want to do.  If you leave it like this I'd just make sure the potential problem case is covered by a unit test. =)

Reply:
We definitely have multiple tests that will fail if this gets messed up haha - I know because I messed it up at one point writing it and tests failed.